src/sctools/groups.py

```diff
@@ -0,0 +1,151 @@
+"""
+Group QC outputs
+
+"""
+
+from crimson import picard
+import argparse
+import os
+import pandas as pd
+
+
+def aggregate_picard_metrics_by_row(file_names, output_name):
+    """
+    pipeline output picard QC metrics at single cell/sample level.
+    This function is called to merge/aggregate QC metrics by metrics type and
+    then merge multiple QC measurement into single matrix file.
+    In this file, column is sample/cell and row is QC metrics
+    :param file_names: list of files generated by Picard
+    :param output_name: output name without suffix'
+    """
+    # initial output
+    metrics = {}
+    d = pd.DataFrame()
+    for file_name in file_names:
+        cell_id = os.path.basename(file_name).split('_qc')[0]
```

Comment:
This feels dependent upon the naming scheme. Can you list the required name format for this tool to work? And, if not standard, which tool's output it should process?

Reply:
added in docstring

Reply:
Got it! with the added docstring, I think I understand now. 

This would be clearer as 
```python3
# trim _qc suffix
basename = os.path.basename(file_name)
if basename.endswith('_qc'):
    cell_id = basename[:-3]
else:
    raise ValueError(
        "QC metric file not properly formatted. Please ensure that file names are formatted as 
        '<cell_id>_qc'."
    )
```